### PR TITLE
Homework 2 - AddEven, GenerateSequence, NwdNww & MaxOfVector

### PR DESCRIPTION
**Pull request by ArturGubala**
Here are my implementations for exercises.

### CODE REVIEW

File: module2/homework/generateSequence/generateSequence.hpp [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
-    return {};
+std::vector<int> generateSequence(int count, int step) 
+{
+    std::vector<int> sequenceResult;
+
+    if (count <= 0)
+    {
+        return sequenceResult;
+    }
+    else
```
Comment (kuchichan) [inline module2/homework/generateSequence/generateSequence.hpp:12]: `else` is redundant here

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 2]
```diff
@@ -2,7 +2,13 @@
 #include <vector>
 #include <limits>
 
-int maxOfVector(const std::vector<int>& vec) {
-    // TODO: Implement me :)
-    return {};
+int maxOfVector(const std::vector<int>& vec)
+{
+    int max = vec.front();
```
Comment (kuchichan) [inline module2/homework/maxOfVector/maxOfVector.hpp:7]: pls check what insteresting things are in `<limits>`

File: module2/homework/addEven/addEven.hpp [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
 
-int addEven(const std::vector<int>& numbers) {
-    // TODO: Your implementation goes here
-    // Below return is only to make this function compile now
-    return -1;
+int addEven(const std::vector<int>& numbers) 
+{
+    int result = 0;
+
+    for (auto number : numbers)
```
Comment (kontrybutor) [inline module2/homework/addEven/addEven.hpp:8]:
```suggestion
    for (const auto & number : numbers)
```
You do not need to modify the ```number``` so you can make it ```const &```

File: module2/homework/nwdNww/nwdNww.hpp [hunk 4]
```diff
@@ -1,11 +1,24 @@
...
-int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+int NWW(int lhs, int rhs) 
+{
+    if (!lhs || !rhs)
+    {
+        return 0;
+    }
+    return abs((lhs * rhs) / NWD(lhs, rhs));
```
Comment (kontrybutor) [inline module2/homework/nwdNww/nwdNww.hpp:23]: I think you can avoid possibly overflow error for large numbers if you divide ```lhs``` by ```NWD(lhs, rhs)``` firstly and then multiply it by ```rhs```

File: module2/homework/generateSequence/generateSequence.hpp (hunk 1 above)
Comment (nauka-programowania-MA) [reply module2/homework/generateSequence/generateSequence.hpp:12]: Please use one style of bracket {}

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 5]
```diff
@@ -2,7 +2,13 @@
 #include <vector>
 #include <limits>
```
Comment (nauka-programowania-MA) [inline module2/homework/maxOfVector/maxOfVector.hpp:3]: if not use remove

File: module2/homework/generateSequence/generateSequence.hpp (hunk 1 above)
Comment (ArturGubala) [reply module2/homework/generateSequence/generateSequence.hpp]: Changed. Thx!

File: module2/homework/addEven/addEven.hpp (hunk 3 above)
Comment (ArturGubala) [reply module2/homework/addEven/addEven.hpp]: Changed. Thx!

File: module2/homework/maxOfVector/maxOfVector.hpp (hunk 2 above)
Comment (ArturGubala) [reply module2/homework/maxOfVector/maxOfVector.hpp]: I checked and changed my implementation. Thx!

File: module2/homework/nwdNww/nwdNww.hpp (hunk 4 above)
Comment (ArturGubala) [reply module2/homework/nwdNww/nwdNww.hpp]: Changed. Thx!

File: module2/homework/maxOfVector/maxOfVector.hpp (hunk 5 above)
Comment (ArturGubala) [reply module2/homework/maxOfVector/maxOfVector.hpp:3]: I changed my implementation and used that library. Thx!

**Comment by ziobron** [comment]
+4 XP addEven
+6 XP NwwNwd
+5 XP maxOfVector
+5 XP generateSequence
+8 XP delivery before 31.05

{ "requirement_ids": [1119, 1631, 1119, 1633, 1119, 1634, 1119, 1635], "points": [2, 4, 2, 6, 2, 5, 2, 5] }

**Pull request closed without merging**